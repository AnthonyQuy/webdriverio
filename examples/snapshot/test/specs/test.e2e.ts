import { expect } from '@wdio/globals'

describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        await browser.url('https://the-internet.herokuapp.com/login')

        await $('#username').setValue('tomsmith')
        await $('#password').setValue('SuperSecretPassword!')
        await $('button[type="submit"]').click()

        async function asyncFunction() {
            await new Promise(resolve => setTimeout(resolve, 1000))
            return { awaited: 'value' }
        }
        await expect(asyncFunction()).toMatchSnapshot()
        await expect({ a: 'a' }).toMatchSnapshot()
        await expect({ a2: 'a' }).toMatchSnapshot()
    })
})
