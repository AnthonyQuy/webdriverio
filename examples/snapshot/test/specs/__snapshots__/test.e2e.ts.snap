// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Login application > should login with valid credentials 1`] = `
Object {
  "awaited": "value",
}
`;

exports[`My Login application > should login with valid credentials 2`] = `
Object {
  "a": "a",
}
`;

exports[`My Login application > should login with valid credentials 3`] = `
Object {
  "a2": "a",
}
`;
